lib/omniauth/strategies/clever.rb

```diff
@@ -42,10 +42,12 @@ def callback_phase
         end
       end
 
-      uid{ raw_info['data']['id'] }
+      uid { raw_info['data']['id'] }
```

Comment:
Ok.   I see that `uid`, `info`, and `extra` are all coming out of the OAuth2 library somehow... I don't exactly understand what they are doing but trust that they work. 

I take it that the big change here is in the raw_info, you are making two calls to the api instead of one and massaging the data into a format you can use.